src/main/kotlin/org/rust/ide/annotator/RsExternalLinterUtils.kt

```diff
@@ -280,7 +280,7 @@ fun RustcSpan.isValid(): Boolean =
     line_end > line_start || (line_end == line_start && column_end >= column_start)
 
 private val ERROR_REGEX: Regex = """E\d{4}""".toRegex()
-private val URL_REGEX: Regex = URLUtil.URL_PATTERN.toRegex()
+private val URL_REGEX: Regex = "(?<!href=['\"])${URLUtil.URL_PATTERN}(?!['\"])".toRegex()
```

Comment:
Are there cases when url that we process with new `URL_REGEX` from original `cargo check`/`check`?
Currently, it seems that our code produces all cases which work incorrectly, for example [here](https://github.com/intellij-rust/intellij-rust/blob/067d839b63540fa8352ffdbb8802af5a17826ea9/src/main/kotlin/org/rust/ide/annotator/RsExternalLinterUtils.kt#L252-L253)
So, probably, the more proper solution is just process only all messages from external linter and not process links generated by our code

Of course, we can handle `<a href="https://%url%">%text%</a>` as well, but I assume `cargo check` and `clippy` just don't generate them